design/pv_backup_info.md

```diff
@@ -0,0 +1,155 @@
+
+Another thing that needs noticing is when describing the backup, there is no generic way to find the PV's snapshot information.
+
+## Goals
+- Create a new metadata file to store backup's PVCs and PVs information and volume data backing up method. The file can be used to let upstream consumers generate a summary.
+- Create a generic way to let the Velero server know how the PV resources are backed up.
+- Create a generic way to let the Velero server find the PV corresponding snapshot information.
+
+## Non Goals
+- Unify how to get snapshot size information for all PV backing-up methods, and all other currently not ready PVs' information.
+
+## High-Level Design
+Create _backup-name_-volumes-info.json metadata file in the backup's repository. This file will be encoded to contain all the PVC and PV information included in the backup. The information covers whether the PV or PVC's data is skipped during backup, how its data is backed up, and the backed-up detail information.
+
+The `restoreItem` function can decode the _backup-name_-volumes-info.json file to determine how to handle the PV resource. 
+
+## Detailed Design
+
+### The VolumeInfo structure
+_backup-name_-volumes-info.json file is an array of structure `VolumeInfo`.
+
+The `VolumeInfo` definition is:
+``` golang
+type VolumeInfo struct {
+    PVCName        string    // The PVC's name. The format should be <namespace-name>/<PVC-name>
+    PVName         string    // The PV name.
+    BackupMethod   string    // The way the volume data is backed up. The valid value includes `VeleroNativeSnapshot`, `PodVolumeBackup`, `CSISnapshot`, `SnapshotDataMover`, and `Skipped`.
+
+    SkippedReason  string    // The reason for the volume is skipped in the backup.
+
+    SnapshotHandle      string       // The actual snapshot ID. It can be the cloud provider's snapshot ID or the file-system uploader's snapshot.
+    Status              string       // The snapshot's final status.
+    Size                int64        // The snapshot corresponding volume size. Some of the volume backup methods cannot retrieve the data by current design, for example, the Velero native snapshot.
+    StartTimestamp      *metav1.Time // Snapshot starts timestamp.
+    CompletionTimestamp *metav1.Time // Snapshot completes timestamp.
+
+    DataMoverInfo       DataMoverInfo
+    NativeSnapshotInfo  VeleroNativeSnapshotInfo
+    PVBInfo             PodVolumeBackupInfo
+}
+
+// DataMoverInfo is used for displaying the snapshot data mover and PodVolumeBackup snapshot status.
+type DataMoverInfo struct {
+    DataMover        string    // The name of the data mover that uploads the snapshot data. The valid values are `kopia` and `restic`. It's useful for file-system backup and snapshot data mover.
+    RetainedSnapshot string    // The ID of the local snapshot.
```

Comment:
This is a snapshot type specific name that identifies the retained snapshot, not always be the ID of snapshots. E.g., for CSI snapshot, this value could be the name of the VSC.

Reply:
How about changing to:
`The name or ID of the snapshot associated object(SAO).`

Reply:
Actually, the DMP knows neither name nor ID or the SAO, so the logics is like -- DMP creates an alias -> DM assigns this alias to the created SAO -> DMP uses this alias later on

Reply:
yes, fine to me